Update and revise README

muhanadz opened the pull request:
As of https://github.com/glencoesoftware/qupath-extension-omero-web/pull/14 the extension was tested with QuPath 0.6.0 and is compatible.

This PR updates the README with the compatible versions.

muhanadz committed:
Update compatibility information for QuPath versions

melissalinkert commented on README.md:
```diff
@@ -19,7 +19,7 @@ where JPEG compression artifacts would be problematic.
 
-The extension is intended for use with QuPath v0.5; it is not compatible with earlier QuPath versions.
+The extension is intended for use with QuPath v0.5 and v0.6; it is not compatible with earlier QuPath versions.
 A *new* QuPath OMERO extension is currently under active development.
```
Maybe outside the scope of this PR, but at some point it may be worth removing this line, and updating the `Docs` section to clarify the difference between what QuPath docs refer to and what is in this repo.

muhanadz replied:
Good point. [ff6ccd7](https://github.com/glencoesoftware/qupath-extension-omero-web/pull/15/commits/ff6ccd7ea648834e9feb0710cd2fe152a171f402) is an attempt at explaining some of the nuances with the different extension versions, some context, updated compatibility version, and pointing to the 0.6 community documentation.

muhanadz committed:
Revise README

Removed outdated compatibility and installation instructions. Updated documentation links and added notes about the relationship to the official QuPath extension.

muhanadz committed:
Update title

melissalinkert reviewed (approved):
Thanks, @muhanadz. This looks good to me.

erindiel commented on README.md:
```diff
@@ -19,30 +19,16 @@ where JPEG compression artifacts would be problematic.
-
-The main documentation for the extension is at https://qupath.readthedocs.io/en/0.5/docs/advanced/omero.html
-
-## Installing
```
Why remove the installation instructions?

erindiel commented on README.md:
```diff
@@ -19,30 +19,16 @@ where JPEG compression artifacts would be problematic.
+This extension is a fork of the original QuPath OMERO extension, maintained by Glencoe Software. 
 
-You might then need to restart QuPath (but not your computer).
+While the [QuPath community extension](https://github.com/qupath/qupath-extension-omero) has recently been updated to support similar features, this repository is maintained to ensure maximum stability and performance for **OMERO Plus** environments using Glencoe Software's microservices.
```
Two questions:
- Is the first sentence suggesting that we are not incorporating changes from the qupath repository into this fork?
- How about putting the OMERO Plus compatibility piece be in the Compatibility section?

erindiel commented on README.md:
```diff
@@ -19,30 +19,16 @@ where JPEG compression artifacts would be problematic.
 
-The output will be under `build/libs`.
-You can drag the jar file on top of QuPath to install the extension.
+> **Note:** The official QuPath documentation refers to the standard community extension. While the user installation, interface, and general workflows (e.g., browsing projects, importing images) are similar, the backend connectivity mechanisms and image retrieval differ in this version.
```
Can you clarify what you mean by "backend connectivity mechanisms" being different? I assume image retrieval is referring to using the OMERO Plus microservices.

erindiel commented on README.md:
```diff
@@ -19,30 +19,16 @@ where JPEG compression artifacts would be problematic.
 
+## Docs
 
-## Building
```
Similarly what is the reason for removing this section?

erindiel reviewed (changes requested):
I have left some in line comments because I think the changes are more extensive than updating version compatibility and should therefore be justified or discussed.

muhanadz committed:
Refine README content and formatting

muhanadz commented:
@erindiel Thanks for the review. Regarding the `Installation` and `Building` sections, in hindsight, removing those details was an oversight on my part. Regarding the phrasing questions:
1. You're right, the phrasing about the community extension was ambiguous. I've reworded this to clarify that we track the community version but optimise this fork for OMERO Plus.
2. I've moved the OMERO Plus specific details into the Compatibility section as suggested.
3. I've clarified 'backend connectivity' to explicitly mention the use of pixel data microservices vs standard Web API.

See [0f7629d](https://github.com/glencoesoftware/qupath-extension-omero-web/pull/15/commits/0f7629dd16de9e5d5cab66a1eb7aef148c4eb42d) for the updated version.

erindiel commented on README.md:
```diff
@@ -1,41 +1,40 @@
-The main documentation for the extension is at https://qupath.readthedocs.io/en/0.5/docs/advanced/omero.html
+General documentation for working with OMERO in QuPath can be found at https://qupath.readthedocs.io/en/0.6/docs/advanced/omero.html.
+
+> **Note:** The official QuPath documentation refers to the standard community extension. While the user installation, interface, and general workflows (e.g., browsing projects, importing images) are similar, this version differs in how it retrieves pixel data (using OMERO Plus microservices when present rather than JSON/JPEG web APIs).
```
```suggestion
> **Note:** The official QuPath documentation refers to the standard community extension. While the user installation, interface, and general workflows (e.g., browsing projects, importing images) are similar, this version differs in how it retrieves pixel data (using OMERO Plus microservices when present rather than the standard OMERO web API).
```

Using the same language as above regarding the `standard OMERO web API`. I am not sure JPEG API makes sense, but open to other language here.

erindiel reviewed (commented):
Looks great, thank you! I had one small text change, but otherwise I think this is ready 👍🏻

muhanadz committed:
Update README.md

Co-authored-by: Erin Diel <41040012+erindiel@users.noreply.github.com>
